src/server_manager/web_app/app.ts

```diff
@@ -920,24 +956,107 @@ export class App {
+    }
+    const previousLimit = this.selectedServer.getAccessKeyDataLimit();
+    if (previousLimit && limit.bytes === previousLimit.bytes) {
+      return;
+    }
+    const serverView = this.appRoot.getServerView(this.appRoot.selectedServer.id);
+    try {
+      await this.selectedServer.setAccessKeyDataLimit(limit);
+      this.appRoot.showNotification(this.appRoot.localize('saved'));
+      serverView.accessKeyDataLimit = this.dataLimitToDisplayDataAmount(limit);
+      this.refreshTransferStats(this.selectedServer, serverView);
+    } catch (error) {
+      console.error(`Failed to set access key data limit: ${error}`);
+      this.appRoot.showError(this.appRoot.localize('error-set-data-limit'));
+      serverView.accessKeyDataLimit = this.dataLimitToDisplayDataAmount(
+          previousLimit || await this.computeDefaultAccessKeyDataLimit(this.selectedServer));
+      serverView.isAccessKeyDataLimitEnabled = !!previousLimit;
+    }
+  }
+
+  private async removeAccessKeyDataLimit() {
+    const serverView = this.appRoot.getServerView(this.appRoot.selectedServer.id);
+    try {
+      await this.selectedServer.removeAccessKeyDataLimit();
+      this.appRoot.showNotification(this.appRoot.localize('saved'));
+      this.refreshTransferStats(this.selectedServer, serverView);
+    } catch (error) {
+      console.error(`Failed to remove access key data limit: ${error}`);
+      this.appRoot.showError(this.appRoot.localize('error-remove-data-limit'));
+      serverView.isAccessKeyDataLimitEnabled = true;
+    }
+  }
+
+  private dataLimitToDisplayDataAmount(limit: server.DataLimit): DisplayDataAmount|null {
+    if (!limit) {
+      return null;
+    }
+    const bytes = limit.bytes;
+    if (bytes >= 10 ** 9) {
+      return {value: Math.round(bytes / (10 ** 9)), unit: 'GB'};
+    }
+    return {value: Math.ceil(bytes / (10 ** 6)), unit: 'MB'};
+  }
+
+  private displayDataAmountToDataLimit(dataAmount: DisplayDataAmount): server.DataLimit {
+    if (!dataAmount) {
+      return null;
+    }
+    if (dataAmount.unit === 'GB') {
+      return {bytes: dataAmount.value * (10 ** 9)};
+    } else if (dataAmount.unit === 'MB') {
+      return {bytes: dataAmount.value * (10 ** 6)};
+    }
+    return {bytes: dataAmount.value};
+  }
+
+  // Compute the suggested data limit based on the server's transfer capacity and number of access
+  // keys.
+  private async computeDefaultAccessKeyDataLimit(
+      server: server.Server, accessKeys?: server.AccessKey[]): Promise<server.DataLimit> {
+    try {
+      // Assume non-managed servers have a data transfer capacity of 1TB.
+      let serverTransferCapacity: server.DataAmount = {terabytes: 1};
+      if (isManagedServer(server)) {
+        serverTransferCapacity = server.getHost().getMonthlyOutboundTransferLimit();
+      }
+      if (!accessKeys) {
+        accessKeys = await server.listAccessKeys();
+      }
+      let dataLimitBytes = serverTransferCapacity.terabytes * (10 ** 12) / accessKeys.length;
```

Comment:
Is it possible for a server to have 0 access keys and error out here? Maybe we want to do (accessKeys.length || 1)